### PR TITLE
EFA-612 Standardize/improve Compose screen state handling

### PR DESCRIPTION
**Pull request by ig-garcia**

### CODE REVIEW

File: k9mail/src/main/java/security/planck/ui/common/compose/input/PasswordInputField.kt [hunk 1]
```diff
@@ -33,19 +33,19 @@ fun PasswordInputField(
...
 ) {
     var passwordVisible by remember { mutableStateOf(false) }
 
     OutlinedTextField(
-        value = passwordState.textState,
+        value = passwordState.text,
         onValueChange = {
-            passwordState.textState = it
-            evaluateError(passwordState)
+            Log.e("EFA-602", "MyScreen text pawwword changed: ${passwordState.text}")
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/ui/common/compose/input/PasswordInputField.kt:44]: typo I guess, even if it is in the logs should be fixed

File: k9mail/src/main/java/security/planck/ui/passphrase/PassphraseViewModel.kt [hunk 2]
```diff
@@ -21,28 +24,42 @@ abstract class PassphraseViewModel(
...
+    /**
+     * Returns the [PassphraseVerificationStatus] according to the format error if any or null.
+     */
+    protected open fun updateAndValidateInput(
+        position: Int,
+        text: String
+    ): PassphraseVerificationStatus? {
+        return passphraseFormatValidator.validatePassphrase(text).let {
+            textFieldStates[position] =
+                textFieldStates[position].copyWith(newText = text, errorStatus = it)
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/ui/passphrase/PassphraseViewModel.kt:48]: how is error status calculated here?

File: k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt [hunk 3]
```diff
@@ -158,8 +140,8 @@ fun RenderManagingAccounts(
         defaultColor = defaultColor,
         errorColor = errorColor,
         successColor = successColor,
-        validateInput = validateNewPassphrase,
-        verifyNewPassphrase = verifyNewPassphrase,
+        validateInput = { validateInput(state.oldPasswordStates.size, it) },
+        verifyNewPassphrase = { validateInput(state.oldPasswordStates.size + 1, it) },
```
Comment (Stereodark) [inline k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt:144]: is the +1 the positional index of the value in the array of values?

**Review by Stereodark — APPROVED** [review]
overall looks fine to me

File: k9mail/src/main/java/security/planck/ui/common/compose/input/PasswordInputField.kt (hunk 1 above)
Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/ui/common/compose/input/PasswordInputField.kt]: logs should be removed, thanks.

File: k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt (hunk 3 above)
Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt:141]: yes, all the values are oldPasswordStates + newPasswordState + newPasswordVerificationState.

File: k9mail/src/main/java/security/planck/ui/passphrase/PassphraseViewModel.kt [hunk 4]
```diff
@@ -21,28 +24,42 @@ abstract class PassphraseViewModel(
...
     }
 
-    protected suspend fun handleFailedVerificationAttempt(accountsWithError: List<String>) {
+    private suspend fun handleFailedVerificationAttempt(accountsWithError: List<String>) {
         failedUnlockAttempts++
         if (failedUnlockAttempts >= MAX_ATTEMPTS_STOP_APP) {
             stateLiveData.value = PassphraseState.TooManyFailedAttempts
         } else {
             if (failedUnlockAttempts >= RETRY_WITH_DELAY_AFTER) {
                 val timeToWait = RETRY_DELAY * 2.0.pow(delayStep).toLong()
```
[the remark this answers is not in the document]
Comment (ig-garcia) [reply k9mail/src/main/java/security/planck/ui/passphrase/PassphraseViewModel.kt]: Simplified it by calculating directly in seconds.

File: k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt (hunk 3 above)
Comment (Stereodark) [reply k9mail/src/main/java/security/planck/ui/passphrase/manage/compose/PassphraseManagementScreen.kt]: great

**Pull request merged**